src/test/server/ServerStatus_test.cpp

```diff
@@ -19,164 +19,542 @@
+
+        //non-secure request
+        {
+            boost::system::error_code ec;
+            beast::http::response_v1<beast::http::string_body> resp;
+            doHTTPRequest(env, yield, false, resp, ec);
             if(! BEAST_EXPECTS(! ec, ec.message()))
                 return;
-            read(ss, sb, resp, ec);
+            BEAST_EXPECT(resp.status == 200);
+        }
+
+        //secure request
+        {
+            boost::system::error_code ec;
+            beast::http::response_v1<beast::http::string_body> resp;
+            doHTTPRequest(env, yield, true, resp, ec);
             if(! BEAST_EXPECTS(! ec, ec.message()))
                 return;
-            BEAST_EXPECT(resp.status == 401);
+            BEAST_EXPECT(resp.status == 200);
         }
-    }
+    };
 
     void
-    testStatusRequest()
+    testTruncatedWSUpgrade(boost::asio::yield_context& yield)
     {
+        testcase("Partial WS upgrade request");
         using namespace jtx;
+        using namespace boost::asio;
+        using namespace beast::http;
         Env env(*this, []()
             {
                 auto p = std::make_unique<Config>();
                 setupConfigForUnitTests(*p);
-                p->section("port_ws").set("protocol", "ws2,wss2");
+                p->section("port_ws").set("protocol", "ws2");
                 return p;
             }());
+
         auto const port = env.app().config()["port_ws"].
             get<std::uint16_t>("port");
-        if(! BEAST_EXPECT(port))
-            return;
+        auto const ip = env.app().config()["port_ws"].
+            get<std::string>("ip");
 
-        using namespace boost::asio;
-        using namespace beast::http;
-        io_service ios;
+        boost::system::error_code ec;
+        response_v1<string_body> resp;
+        auto req = makeWSUpgrade(*ip, *port);
+
+        //truncate the request message to near the value of the version header
+        auto req_string = boost::lexical_cast<std::string>(req);
+        req_string.erase(req_string.find_last_of("13"), std::string::npos);
+
+        io_service& ios = get_io_service();
         ip::tcp::resolver r{ios};
         beast::streambuf sb;
-        response_v1<string_body> resp;
+
+        auto it =
+            r.async_resolve(
+                ip::tcp::resolver::query{*ip, to_string(*port)}, yield[ec]);
+        if(! BEAST_EXPECTS(! ec, ec.message()))
+            return;
+
+        ip::tcp::socket sock{ios};
+        async_connect(sock, it, yield[ec]);
+        if(! BEAST_EXPECTS(! ec, ec.message()))
+            return;
+        async_write(sock, boost::asio::buffer(req_string), yield[ec]);
+        if(! BEAST_EXPECTS(! ec, ec.message()))
+            return;
+        // since we've sent an incomplete request, the server will
+        // keep trying to read until it gives up (by timeout)
+        async_read(sock, sb, resp, yield[ec]);
+        BEAST_EXPECT(ec);
+        BEAST_EXPECTS(ec.message() == "End of file", ec.message());
```

Comment:
ok - thanks. I'll just change the test to make sure the message is not empty since this gets into the realm of OS specific err codes. Did any other ec.message() checks fail?